tests/k256_auth.json

```diff
@@ -0,0 +1,27 @@
+[
+  {
```

Comment:
I couldn't get the `auth` mode vectors to work. I was getting failing tests. I'm not sure if it's because the format of the vectors is slightly different or if I actually have an error in the implementation somewhere. I checked in this single auth vector to see if anything jumped out at you that might be wrong with it.

Reply:
`derive_key_pair` appears to generate the wrong keys based on `ikmS`. That's not too much code but something seems wrong in there. Or might also be that the test vector is wrong. When you checked your code here you could also check the code that gnereated this to make sure it's not doing something wrong.